### PR TITLE
fix: add a 'jx edit requirements' step to edit the requirements file

### LINKED ISSUE #5655: provide a `jx requirements` command to create or edit the `jx-requirements.yml`
lets provide a simple CLI command to let folks configure the `jx-requirements.yml` using [bash completion](https://jenkins-x.io/docs/using-jx/faq/)

e.g.

`jx requirements --domain foo.bar --provider eks` 

which then sets the required fields in the `jx-requirements.yml` file

### PR DESCRIPTION
**Pull request by jstrachan**
lets add a simple step to let folks use bash completion to create/modify the `jx-requirements.yml` file

fixes #5655

Signed-off-by: James Strachan <james.strachan@gmail.com>

### CODE REVIEW

**Comment by abayer** [comment]
Playing devil's advocate - what's the value for this over editing with a schema?

**Comment by jstrachan** [comment]
@abayer it was a response to someone who wanted to modify the file via the CLI + command line arguments + bash completion on the slack channel today. Its more a choice really; use a schema aware IDE - or use CLI + bash completion - they both do the same thing, modify the same underlying file

**Comment by jstrachan** [comment]
@abayer I guess another way of saying that is - this is like the migration of `jx install` to something boot-ish - so there's lots of CLI arguments with nicer validation available

File: pkg/cmd/requirements/requirements.go [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+	requirementsLong = templates.LongDesc(`
+		Creates or edits the local 'jx-requirements.yml file for 'jx boot'
+`)
+
+	requirementsExample = templates.Examples(`
+		# creates or modifies the local 'jx-requirements.yml' file used for 'jx boot'
+		jx requirements --domain foo.com --tls --provider eks
+
+		# Opens a terminal in the cheese container in the latest pod in the foo deployment
+		jx rsh -c cheese foo
```
Comment (abayer) [inline pkg/cmd/requirements/requirements.go]: This ain't quite right. =)

Comment (jstrachan) [reply pkg/cmd/requirements/requirements.go]: thanks @abayer :)

**Comment by abayer** [comment]
Ok. Would it make more sense to have `jx requirements edit` and a corresponding `jx requirements get` or something to get current values from the requirements?

**Comment by jstrachan** [comment]
what does `jx requirements get` do? the requirements are already in the `jx-requirements.yml` file on disk? 

I did wonder about `jx create requirements / jx edit requirements` but figured it was maybe easier to just have 1 command

**Comment by abayer** [comment]
Well, if we're adding a command to edit the file on disk, why not add a command to fetch specific values from the file on disk? =) Just a thought.

**Comment by jstrachan** [comment]
so `jx get requirements` == `cat jx-requirements.yml`? :)

**Comment by abayer** [comment]
I meant more like `jx get requirements --key=foo` or whatever. =)

**Comment by abayer** [comment]
@jstrachan You should look at the logs. =) Legit failure:

```
    requirements_test.go:89: running test bbs
panic: unable to redefine 'r' shorthand in "requirements" flagset: it's already used for "registry" flag [recovered]
	panic: unable to redefine 'r' shorthand in "requirements" flagset: it's already used for "registry" flag
goroutine 70 [running]:
testing.tRunner.func1(0xc00064ac00)
	/usr/local/go/src/testing/testing.go:830 +0x392
panic(0x23656a0, 0xc000440190)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/spf13/pflag.(*FlagSet).AddFlag(0xc00064ad00, 0xc0002ebb80)
	/home/jenkins/go/pkg/mod/github.com/spf13/pflag@v1.0.3/flag.go:862 +0x6d8
github.com/spf13/pflag.(*FlagSet).VarPF(0xc00064ad00, 0x31ff400, 0xc000163ca8, 0x281d7c6, 0x6, 0x2818e44, 0x1, 0x284c9b5, 0x1b, 0xc0002ebae0)
	/home/jenkins/go/pkg/mod/github.com/spf13/pflag@v1.0.3/flag.go:819 +0x10b
github.com/spf13/pflag.(*FlagSet).VarP(...)
	/home/jenkins/go/pkg/mod/github.com/spf13/pflag@v1.0.3/flag.go:825
github.com/spf13/pflag.(*FlagSet).StringVarP(0xc00064ad00, 0xc000163ca8, 0x281d7c6, 0x6, 0x2818e44, 0x1, 0x0, 0x0, 0x284c9b5, 0x1b)
	/home/jenkins/go/pkg/mod/github.com/spf13/pflag@v1.0.3/string.go:42 +0xaf
github.com/jenkins-x/**/pkg/cmd/requirements.NewCmdRequirements(0xc000522300, 0xc00064ac00)
	/workspace/source/pkg/cmd/requirements/requirements.go:99 +0xabe
github.com/jenkins-x/**/pkg/cmd/requirements_test.TestCmdRequirements(0xc00064ac00)
	/workspace/source/pkg/cmd/requirements/requirements_test.go:102 +0x96c
testing.tRunner(0xc00064ac00, 0x2946bf0)
	/usr/local/go/src/testing/testing.go:865 +0xc0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:916 +0x35a
FAIL	github.com/jenkins-x/**/pkg/cmd/requirements	0.147s
```

File: pkg/cmd/cmd.go [hunk 2]
```diff
@@ -105,6 +106,7 @@ func NewJXCommand(f clients.Factory, in terminal.FileReader, out terminal.FileWr
 	installCommands = append(installCommands, findCommands("cluster", updateCommands)...)
 	installCommands = append(installCommands, findCommands("jenkins token", createCommands, deleteCommands)...)
 	installCommands = append(installCommands, initcmd.NewCmdInit(commonOpts))
+	installCommands = append(installCommands, requirements.NewCmdRequirements(commonOpts))
```
Comment (pmuir) [inline pkg/cmd/cmd.go]: Shouldn't this be `jx edit requirements`?

Comment (pmuir) [reply pkg/cmd/cmd.go]: I was expecting something a bit more `jx edit userroles` actually.

Comment (jstrachan) [reply pkg/cmd/cmd.go]: renamed to `jx edit requirements`

**Pull request merged**